File: src/main/javascript/processing/bugfest/Bacterium.js
class Bacterium extends Animal {
  constructor(designModeActive, designImage, columnWidth, rowHeight) {
		super();

    this.designModeActive = designModeActive;
		this.designImage = designImage;
    this.columnWidth = columnWidth;
    this.rowHeight = rowHeight;

    this.bacteriumPoints = [];

		this.colorBlack = color(0);
    this.colorRed = color(231, 24, 9);
		this.colorOrange = color(255, 165, 0);
		this.colorYellow = color(255, 231, 0);
		this.colors = [this.colorRed, this.colorOrange, this.colorYellow];
	}

  initializePoints() {
    this.bacteriumPoints = [
      [
        // Body points.
				new p5.Vector(0.34 * columnWidth, 0.40 * rowHeight),
				new p5.Vector(0.42 * columnWidth, 0.66 * rowHeight),
				new p5.Vector(0.50 * columnWidth, 0.74 * rowHeight),
				new p5.Vector(0.56 * columnWidth, 0.75 * rowHeight),
				new p5.Vector(0.63 * columnWidth, 0.70 * rowHeight),
				new p5.Vector(0.66 * columnWidth, 0.58 * rowHeight),
				new p5.Vector(0.62 * columnWidth, 0.47 * rowHeight),
				new p5.Vector(0.56 * columnWidth, 0.31 * rowHeight),
				new p5.Vector(0.50 * columnWidth, 0.25 * rowHeight),
				new p5.Vector(0.44 * columnWidth, 0.24 * rowHeight),
				new p5.Vector(0.38 * columnWidth, 0.26 * rowHeight),
				new p5.Vector(0.35 * columnWidth, 0.32 * rowHeight),
				new p5.Vector(0.33 * columnWidth, 0.40 * rowHeight),
      ],
			[
				new p5.Vector(0.34 * columnWidth, 0.40 * rowHeight),
				new p5.Vector(0.29 * columnWidth, 0.38 * rowHeight),
				new p5.Vector(0.24 * columnWidth, 0.38 * rowHeight),
				new p5.Vector(0.20 * columnWidth, 0.39 * rowHeight),
				new p5.Vector(0.16 * columnWidth, 0.37 * rowHeight),
				new p5.Vector(0.14 * columnWidth, 0.35 * rowHeight),
			],
			[
				new p5.Vector(0.38 * columnWidth, 0.56 * rowHeight),
				new p5.Vector(0.34 * columnWidth, 0.54 * rowHeight),
				new p5.Vector(0.29 * columnWidth, 0.58 * rowHeight),
				new p5.Vector(0.24 * columnWidth, 0.61 * rowHeight),
				new p5.Vector(0.19 * columnWidth, 0.61 * rowHeight),
			],
			[
				new p5.Vector(0.59 * columnWidth, 0.35 * rowHeight),
				new p5.Vector(0.63 * columnWidth, 0.32 * rowHeight),
				new p5.Vector(0.69 * columnWidth, 0.31 * rowHeight),
				new p5.Vector(0.73 * columnWidth, 0.33 * rowHeight),
				new p5.Vector(0.76 * columnWidth, 0.31 * rowHeight),
				new p5.Vector(0.78 * columnWidth, 0.26 * rowHeight),
			],
			[
				new p5.Vector(0.64 * columnWidth, 0.49 * rowHeight),
				new p5.Vector(0.69 * columnWidth, 0.50 * rowHeight),
				new p5.Vector(0.74 * columnWidth, 0.49 * rowHeight),
				new p5.Vector(0.78 * columnWidth, 0.49 * rowHeight),
				new p5.Vector(0.86 * columnWidth, 0.51 * rowHeight),
			],
    ];
  }

  draw() {
    if (this.designModeActive) {
      image(this.designImage, this.columnWidth / 2, this.rowHeight / 2, this.columnWidth, this.rowHeight);
    }
    else {
      this.initializePoints();
      this.randomizePoints(this.bacteriumPoints, 0.02);

      // Draw the body.

      fill(this.colors[int(random(this.colors.length))]);

      this.drawShape(this.bacteriumPoints[0], this.LEFT_SIDE, CLOSE);
      // this.drawShape(this.bacteriumPoints[0], this.RIGHT_SIDE, CLOSE);

      // Draw the legs.
      stroke(0);
      strokeWeight(1);
      fill(255);

			for (let legIndex = 0; legIndex < 4; legIndex++) {
        this.drawShape(this.bacteriumPoints[legIndex + 1], this.LEFT_SIDE, CLOSE);
        // this.drawShape(this.bacteriumPoints[legIndex + 1], this.RIGHT_SIDE, CLOSE);
			}
		}
  }
}

File: src/main/kotlin/com/github/freekdb/processing/bugfest/Bacterium.kt
package com.github.freekdb.processing.bugfest

import processing.core.PApplet
import processing.core.PImage
import processing.core.PVector

class Bacterium(
    private val parentApplet: PApplet, columnWidth: Int, rowHeight: Int, designImage: PImage?
) : Organism(parentApplet, columnWidth, rowHeight, false, designImage) {
    init {
        this.setParent(parentApplet)
    }

    private val colorBlack = parentApplet.color(0)
    private val colorRed = parentApplet.color(231, 24, 9)
    private val colorOrange = parentApplet.color(255, 165, 0)
    private val colorYellow = parentApplet.color(255, 231, 0)
    private val colors = listOf(colorRed, colorOrange, colorYellow)

    private val bacteriumPoints = listOf(
        mutableListOf(
            // Body points.
            PVector(0.34f * columnWidth, 0.40f * rowHeight),
            PVector(0.42f * columnWidth, 0.66f * rowHeight),
            PVector(0.50f * columnWidth, 0.74f * rowHeight),
            PVector(0.56f * columnWidth, 0.75f * rowHeight),
            PVector(0.63f * columnWidth, 0.70f * rowHeight),
            PVector(0.66f * columnWidth, 0.58f * rowHeight),
            PVector(0.62f * columnWidth, 0.47f * rowHeight),
            PVector(0.56f * columnWidth, 0.31f * rowHeight),
            PVector(0.50f * columnWidth, 0.25f * rowHeight),
            PVector(0.44f * columnWidth, 0.24f * rowHeight),
            PVector(0.38f * columnWidth, 0.26f * rowHeight),
            PVector(0.35f * columnWidth, 0.32f * rowHeight),
            PVector(0.33f * columnWidth, 0.40f * rowHeight),
        ),
        mutableListOf(
            PVector(0.34f * columnWidth, 0.40f * rowHeight),
            PVector(0.29f * columnWidth, 0.38f * rowHeight),
            PVector(0.24f * columnWidth, 0.38f * rowHeight),
            PVector(0.20f * columnWidth, 0.39f * rowHeight),
            PVector(0.16f * columnWidth, 0.37f * rowHeight),
            PVector(0.14f * columnWidth, 0.35f * rowHeight),
        ),
        mutableListOf(
            PVector(0.38f * columnWidth, 0.56f * rowHeight),
            PVector(0.34f * columnWidth, 0.54f * rowHeight),
            PVector(0.29f * columnWidth, 0.58f * rowHeight),
            PVector(0.24f * columnWidth, 0.61f * rowHeight),
            PVector(0.19f * columnWidth, 0.61f * rowHeight),
        ),
        mutableListOf(
            PVector(0.59f * columnWidth, 0.35f * rowHeight),
            PVector(0.63f * columnWidth, 0.32f * rowHeight),
            PVector(0.69f * columnWidth, 0.31f * rowHeight),
            PVector(0.73f * columnWidth, 0.33f * rowHeight),
            PVector(0.76f * columnWidth, 0.31f * rowHeight),
            PVector(0.78f * columnWidth, 0.26f * rowHeight),
        ),
        mutableListOf(
            PVector(0.64f * columnWidth, 0.49f * rowHeight),
            PVector(0.69f * columnWidth, 0.50f * rowHeight),
            PVector(0.74f * columnWidth, 0.49f * rowHeight),
            PVector(0.78f * columnWidth, 0.49f * rowHeight),
            PVector(0.86f * columnWidth, 0.51f * rowHeight),
        ),
    )

//    override fun drawOrganism() {
//        randomizePoints(bacteriumPoints, 0.02f)
//
//        // Draw the body.
//        parentApplet.fill(colors[parentApplet.random(colors.size.toFloat()).toInt()])
//        drawShape(this.bacteriumPoints[0], LEFT_SIDE, CLOSE)
//
//        // Draw the legs.
//        parentApplet.stroke(colorBlack)
//        parentApplet.strokeWeight(1f)
//        parentApplet.fill(255)
//
//        for (legIndex in 0..3) {
//            drawShape(bacteriumPoints[legIndex + 1], LEFT_SIDE, CLOSE, LineStyle.LINE_VERTEX)
//        }
//    }

    override fun getOrganismData(): OrganismData {
        return OrganismData(emptyList())
    }
}
